### PR TITLE
feat(tips on v2): Add community tip directly to v2 data transaction PE-1453

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds the community tip to v2 data transactions and consolidates handling the community tip to the `PstService` class.

### CODE REVIEW

**Review by jdaev — APPROVED** [review]
👍🏽

**Review by thiagocarvalhodev — APPROVED** [review]

File: lib/blocs/upload/cost_estimate.dart [hunk 1]
```diff
@@ -52,60 +45,26 @@ class CostEstimate {
         arweaveService: arweaveService);
 
     final bundlePstFee = await pstService.getPSTFee(dataItemsCost);
+    final v2FilesPstFee = v2FilesUploadCost <= BigInt.zero
```
Comment (arielmelendez) [inline lib/blocs/upload/cost_estimate.dart:48]: Less than zero should arguably be an error.

Comment (arielmelendez) [reply lib/blocs/upload/cost_estimate.dart:48]: No need to change this though. I think it's good defense even if we expect something to throw upstream.

Comment (arielmelendez) [reply lib/blocs/upload/cost_estimate.dart:48]: Actually wouldn't it be cleaner just just let `await pstService.getPSTFee(v2FilesUploadCost);` handle this logic?

Comment (fedellen) [reply lib/blocs/upload/cost_estimate.dart:48]: This can be 0 when all files are bundled. It is just for estimating totalPrice

**Review by arielmelendez — APPROVED** [review]
Looks good. Issued one minor suggestion.

**Pull request merged**